src/util/battery/battery.cpp

```diff
@@ -0,0 +1,49 @@
+#include "util/battery/battery.h"
+
+#ifdef Q_OS_LINUX
+#include "util/battery/batterylinux.h"
+#elif defined(Q_OS_WIN)
+#include "util/battery/batterywindows.h"
+#elif defined(Q_OS_MAC)
+#include "util/battery/batterymac.h"
+#endif
+
+// interval (in ms) of the timer which calls update()
+const int kiUpdateInterval = 5000;
+
+Battery::Battery(QObject* parent)
+        : QObject(parent),
+          m_chargingState(UNKNOWN),
+          m_dPercentage(0.0),
+          m_iMinutesLeft(0),
+          timer(this) {
+    connect(&timer, SIGNAL(timeout()), this, SLOT(update()));
+    timer.start(kiUpdateInterval);
```

Comment:
I added GuiTickTimer so we have a drop-in QTimer replacement that uses the GuiTick 50ms signal.